### PR TITLE
[LAB-10] Add HTTPS support for ZAT

### PR DESCRIPTION
**Pull request by adammw**
:koala:

Add HTTPS support for ZAT

```
$ zat serve --ssl-cert /opt/boxen/config/nginx/ssl/zendesk.dev.crt --ssl-key
/opt/boxen/config/nginx/ssl/zendesk.dev.key
== Sinatra/1.4.5 has taken the stage on 4567 for development with backup from Thin
Thin web server (v1.6.2 codename Doc Brown)
Maximum connections set to 1024
Listening on localhost:4567, CTRL+C to stop
== ZAT Server is running
You may now start using ZAT by appending ?zat=https://localhost:4567/app.js to your Zendesk URL
^CStopping ...
Stopping ...
== Sinatra has ended his set (crowd applauds)
```

/cc @zendesk/quokka
### References
- Jira link: https://zendesk.atlassian.net/browse/LAB-10
### Risks
- None

### CODE REVIEW

File: lib/zendesk_apps_tools/command.rb [hunk 1]
```diff
@@ -150,5 +191,57 @@ def setup_path(path)
...
+      File.write(localhost_ssl_cert_path, certificate.self_signed_pem)
+      File.write(localhost_ssl_key_path, certificate.private_key)
+    end
+
+    def trust_certificate
+      puts "Trusting certificate..."
+      if OS.mac?
+        exec_cmd("sudo security add-trusted-cert -d -r trustRoot -k /Library/Keychains/System.keychain \"#{localhost_ssl_cert_path}\"")
+      elsif OS.windows?
+        exec_cmd("certutil.exe -addstore -user root \"#{localhost_ssl_cert_path}\"")
```
Comment (princemaple) [inline lib/zendesk_apps_tools/command.rb]: where is this `.exe` file from?

Comment (adammw) [inline lib/zendesk_apps_tools/command.rb]: https://technet.microsoft.com/en-us/library/cc732443.aspx I think, but it's probably not available on all versions of Windows. I'm tempted to undo this part of the PR and make it a manual step, thoughts @princemaple ?

Comment (princemaple) [inline lib/zendesk_apps_tools/command.rb]: Manual or drop Windows support (of this functionality) for now. I'm pretty sure not many Ruby developers use Windows. From my own experience, working on Ruby on Windows is quite painful.

Comment (adammw) [inline lib/zendesk_apps_tools/command.rb]: :haircut: I've got it to fallback if those commands don't exist on the system /cc @princemaple

**Comment by adammw** [comment]
:haircut: /cc @zendesk/quokka

**Comment by princemaple** [comment]
Let me teach you how to be a jerk in code review.
comparing_adammw_https-zat_-_zendesk_apps_tools_-_code_climate

**Comment by adammw** [comment]
@princemaple I agree, any ideas how I can refactor some of this code into a separate module?

**Comment by princemaple** [comment]
you could take commands out and have a separate module for every command. and more separate modules for helper methods

**Comment by Archinowsk** [comment]
Any updates on this? I would give it a try myself but Ruby is not my strong suite.

**Comment by hoangtrvu** [comment]
Closing due to low activity.

**Pull request closed without merging**